frappe/database/mariadb/schema.py

```diff
@@ -102,11 +103,24 @@ def alter(self):
 				if index_record := frappe.db.get_column_index(self.table_name, col.fieldname, unique=False):
 					drop_index_query.append(f"DROP INDEX `{index_record.Key_name}`")
 
+		for col in self.change_nullability:
+			current_column = self.current_columns.get(col.fieldname.lower())
+			if col.not_nullable:
+				default_value = get_not_null_defaults(col.fieldtype)
+				try:
+					table = frappe.qb.DocType(self.doctype)
+					frappe.qb.update(table).set(col.fieldname, default_value).where(
+						getattr(table, col.fieldname).isnull()
+					).run()
+				except Exception as e:
```

Comment:
Brainstorm what are the failure modes here? 
- Db constraints causing it to fail - don't allow invalid db constraints.